test/services/EventReportServiceSpec.scala

```diff
@@ -352,39 +355,140 @@ class EventReportServiceSpec extends AsyncWordSpec with Matchers with MockitoSug
   }
 
   "submitEventDeclarationReport" must {
-    "return valid response" in {
+    "return valid response where there are changes" in {
+      val (userAnswers, submitEventDeclarationReportSuccessResponseETMP) = super[GeneratorAPI1828].generateUserAnswersAndPOSTBody.sample.value
+      when(mockEventReportConnector.submitEventDeclarationReport(
+        ArgumentMatchers.eq(pstr),
+        any())(any(), any(), any()))
+        .thenReturn(Future.successful(HttpResponse.apply(
+          status = OK,
+          json = payload,
+          headers = Map.empty)))
+      eventReportService.submitEventDeclarationReport(pstr, userAnswers)(implicitly, implicitly, implicitly).map { _ =>
+        verify(mockEventReportConnector, times(1)).submitEventDeclarationReport(ArgumentMatchers.eq(pstr),
+          ArgumentMatchers.eq(submitEventDeclarationReportSuccessResponseETMP))(any(), any(), any())
+        assert(true)
+      }
+    }
+    "return valid an invalid response where an 1829 payload is passed" in {
+      val userAnswers= super[GeneratorAPI1829].generateUserAnswersAndPOSTBody.sample.value._1
       when(mockEventReportConnector.submitEventDeclarationReport(
         ArgumentMatchers.eq(pstr),
-        ArgumentMatchers.eq(submitEventDeclarationReportSuccessResponse))(any(), any(), any()))
+        any())(any(), any(), any()))
         .thenReturn(Future.successful(HttpResponse.apply(
           status = OK,
-          json = submitEventDeclarationReportSuccessResponse,
+          json = payload,
           headers = Map.empty)))
-      eventReportService.submitEventDeclarationReport(pstr, submitEventDeclarationReportSuccessResponse)(implicitly, implicitly, implicitly).map { resultJsValue =>
-        resultJsValue mustBe submitEventDeclarationReportSuccessResponse
+      eventReportService.submitEventDeclarationReport(pstr, userAnswers)(implicitly, implicitly, implicitly).map { _ =>
+        verify(mockEventReportConnector, times(1)).submitEventDeclarationReport(ArgumentMatchers.eq(pstr),
+          any())(any(), any(), any())
+        assert(true)
+      }
+    }
+    "return a 417 error response when there is nothing to submit" in {
+      val (userAnswers, submitEventDeclarationReportSuccessResponseETMP) = super[GeneratorAPI1828].generateUserAnswersAndPOSTBody.sample.value
+      when(mockEventReportConnector.submitEventDeclarationReport(
+        ArgumentMatchers.eq(pstr),
+        any())(any(), any(), any()))
+        .thenReturn(Future.failed(new BadRequestException("Test"))) //TODO: This needs to change when Sanjay gets back with actual error response
```

Comment:
Do we have a ticket to keep track of the progress for these TODOs? If so we can add the ticket number to this and the following comments so that we don't forget to implement.

Reply:
Removed as no longer relevant